src/libsyntax/parse/parser/expr.rs

```diff
@@ -844,29 +843,29 @@ impl<'a> Parser<'a> {
                 return self.parse_block_expr(None, lo, BlockCheckMode::Default, attrs);
             }
             token::BinOp(token::Or) | token::OrOr => {
-                return self.parse_lambda_expr(attrs);
+                return self.parse_closure(attrs);
```

Comment:
:(

We should imo use lambda everywhere; the closure is the environment of a lambda term.

Also, the convention is to always include `expr` somewhere in the method names.

Reply:
I agree, but for the sake of consistency, it was easier to change those two... I'll add `expr` back though.

Reply:
Alright fine.